### DISCUSSION TITLE
Method to clear chart

### DISCUSSION THREAD

**Discussion by LeeWeeder** [patrykandpatrick/vico#818] [Questions] [answered]
### Question

I have used this library for quite some time and made it work for my use case, however, are there any methods or ways to clear the chart after the modelProducer has been attached? As creating a new `CartesianChartModelProducer` is out of the table. 
The problem I had is when the user deletes all the data from the list, the last deleted data will be still on the chart. It goes that way because before calling `runTransaction`, I first check whether the data is empty because empty series isn't allowed. Thus the chart will never recompose with empty data.

> [!NOTE]
> I am already using `placeholder` but it only works if the `runTransaction` wasn't initially called (during the lifetime of the ViewModel as in my implementation). (Correct me if I'm wrong.) Thus, when restarting the app, the chart loads correctly. 

### Vico version(s)

2.0.0-alpha.26

### UI framework(s)

Jetpack Compose

**Comment by patrickmichalik** [comment, marked as answer]
Hello! You can do this by running a `Transaction` that removes all data. The placeholder will appear.

```kt
modelProducer.runTransaction { }
```

Here, I’d suggest implementing this by moving the emptiness check to the `runTransaction` block, so `runTransaction` itself is called even when there’s no data, but the scope functions are invoked conditionally. The logic may look something like this:

```kt
modelProducer.runTransaction {
  if (data.isEmpty()) return@runTransaction
  // ...
}
```

**Reply by LeeWeeder** [reply]
Whew! It's that simple? Thank you!

**Reply by patrickmichalik** [reply]
You’re welcome!
